README.md

```diff
@@ -16,18 +16,24 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 
 # Abstract
 
-This document describes PomoFlow, a dataflow runtime for PomoDB. This design is especially suited for incrementalizing programs to efficiently compute over deltas to the input EDB.
+This document describes PomoFlow, a [dataflow] query runtime for PomoDB. It is intended for incrementalizing programs to efficiently compute over deltas from an extensional database.
 
 # 1. Introduction
 
-Most relational query runtimes implement a variant of semi-naive evaluation, where queries are run over the course of multiple iterations, with each iteration refining the result toward a fixed point. Such designs can efficiently compute views over a fixed database, but are insufficient for incrementalizing evaluation over a database that changes over time.
+## 1.1 Motivation
 
-This document describes an alternative runtime, built on dataflow, which represents programs as circuits whose vertices and edges correspond to computations over streams of data. These circuits can be incrementalized to instead operate over deltas, with the results being combined back into a materialized view.
+Most relational query runtimes implement a variant of [semi-naive evaluation], where queries are iteratively refined until they reach a fixed point. Such designs can efficiently compute views over a fixed database, but are insufficient for incrementalizing evaluation over a database that changes over time. Since semi-naive evaluation requires evaluating the same data many times, it can be inefficient for many applications, especially when realtime operation is desired.
```

Comment:
Nitpick: it might be a little confusing to say that semi-naive evaluation requires evaluating the same data multiple times, because in the general case, it doesn't, and that's the big benefit that it offers over naive evaluation. The issue is that semi-naive evaluation can't handle non-monotonic updates, without rolling back the database and re-evaluating.

Reply:
Ah indeed I could phrase this more clearly. Seminaive would need to rerun the same query to do e.g. a collaborative text editor

Reply:
@QuinnWilton how about something like this:

> Most datalog query runtimes implement a variant of [semi-naive evaluation], where queries are iteratively refined until they reach a fixed point. Such designs can efficiently compute views over a fixed database, but are insufficient for incrementalizing evaluation over a database that changes over time. Since semi-naive evaluation of nonmonotonic queries requires evaluation on changes to the input, it can be inefficient for many applications, especially when real time operation is desired.

Reply:
> Since semi-naive evaluation of nonmonotonic queries requires re-evaluation on changes to the input

Assuming you meant re-evaluation there, SGTM :)